### PR TITLE
wget: upgrade version 1.21.2 -> 2.1.0

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade version from 1.21.2 -> 2.1.0

PS: I had a chat with Jon w.r.t the name of spec being `wget` instead of `wget2`. That causes a lot of hacks to be put in the spec, which IMO is not worth it.
There is a `Provides: wget`; which ensures that `tdnf install wget` works properly.

The tests on the package are failing, but my proposal is that we check it in; The tests would be checked in a subsequent PR as they may need additional packages.
Atleast by this way, it would get some amount of testing meanwhile.

###### Change Log  

- Adapt spec from Fedora 40.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build and verification.

Pipeline build id: 
- BuddyBuild: [PR-7672](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=498944&view=results)
- FullBuild: [link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=498945&view=results)

BuddyBuild Post addressing review comments: [PR-7672](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=516053&view=results)

### CODE REVIEW

File: SPECS/wget2/wget2.spec [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+%package devel
+Summary:        Libraries and header files needed for using wget2 libraries
+Requires:       %{name}-libs%{?_isa} = %{version}-%{release}
+
+%description devel
+Development libraries and headers needed for building applications to
+use functionality from GNU Wget2.
+
+%if %{as_wget}
+%package wget
```
Comment (kanikanema) [inline SPECS/wget2/wget2.spec:79]: Should this be `package -n wget`

File: SPECS/wget2/wget2.spec [hunk 2]
```diff
@@ -0,0 +1,210 @@
+%global as_wget 1
+%global somajor 2
+
+Name:           wget2
```
Comment (jslobodzian) [inline SPECS/wget2/wget2.spec:4]: Please make this just wget and remove the old wget.

Comment (mfrw) [reply SPECS/wget2/wget2.spec]: ack

**Comment by mfrw** [comment]
Local Verification:
image

**Review by kanikanema — COMMENTED** [review]
LGTM. Some minor comments.

File: SPECS/wget/wget.spec [hunk 3]
```diff
@@ -30,37 +83,64 @@ downloading of files from the Web.
...
+%{_includedir}/wget.h
+%{_includedir}/wgetver.h
+%{_libdir}/libwget*.so
+%{_libdir}/pkgconfig/libwget.pc
+%{_mandir}/man3/libwget*.3*
+
 %changelog
+* Wed Feb 28 2024 Muhammad Falak <mwani@microsoft.com> - 2.1.0-1
+- Switch wget from 1.x to 2.x
+- Add a provides for wget
```
[the remark this answers is not in the document]
Comment (mfrw) [reply SPECS/wget/wget.spec:140]: Yes thanks. will remove

**Pull request merged**